### PR TITLE
Webreporter API: Save Run Data for third party webreporter

### PR DESCRIPTION
**Pull request by lpgeiger**
Persist the raw data from the mocha eventemitters. Proposed solution for https://github.com/DispatchMe/meteor-mocha/issues/9

This can be used in a practicalmeteor:mocha type web reporter.

Client side needs some work. Circular references and events generate $ref and $events keys. The $ is not allowed in mongo. Getting rid of '$' sign is a hack, must be a better way.

### CODE REVIEW

**Comment by keyscores** [comment]
@aldeed  see here a prototype of a web reporting using mongo for reporting. 
With this working maybe you can see how best to divide into different packages. I don't think it adds that much overhead beyond the other changes proposed.

File: webreporter/webreporter.js [hunk 1]
```diff
@@ -0,0 +1,223 @@
+import '../lib/collections';
+import { Template } from 'meteor/templating'
+import { Session } from 'meteor/session'
+
+if(Meteor.isClient){
+  Meteor.startup(function(){
+    Session.set("filterState", 'all')
+
+
+    // TODO: should run again when page refreshes,
```
Comment (keyscores) [inline webreporter/webreporter.js]: @aldeed  I'm stuck on something. If I place all the server code in a 'runAllTests' method, subsequent calls after the initial test run, raise the error in mocha:
TypeError: Cannot read property 'call' of undefined.

I'm not familiar enough with mocha-core to know why this is happening. Any clues?

Comment (aldeed) [inline webreporter/webreporter.js]: @keyscores I'm not sure offhand and I'm heading out on vacation, but remind me to take a closer look when I get back.

Comment (lpgeiger) [inline webreporter/webreporter.js]: @aldeed following up on this. I'd like to rerun all the tests programmatically. This means putting the test runner in a method. But I'm getting strange errors such as `TypeError: Cannot read property 'call' of undefined.`.

**Comment by lpgeiger** [comment]
@aldeed following up on this. I'd like to rerun all the tests programmatically. This means putting the test runner in a method. But I'm getting strange errors such as TypeError: Cannot read property 'call' of undefined...
Any ideas?

**Pull request closed without merging**